src/test/HostActivationTests/FrameworkResolution/RollForwardMultipleFrameworks.cs

```diff
@@ -0,0 +1,544 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.Cli.Build;
+using Microsoft.DotNet.Cli.Build.Framework;
+using System;
+using Xunit;
+
+namespace Microsoft.DotNet.CoreSetup.Test.HostActivation.FrameworkResolution
+{
+    public class RollForwardMultipleFrameworks :
+        FrameworkResolutionBase,
+        IClassFixture<RollForwardMultipleFrameworks.SharedTestState>
+    {
+        private const string MiddleWare = "MiddleWare";
+        private const string AnotherMiddleWare = "AnotherMiddleWare";
+        private const string HighWare = "HighWare";
+
+        private SharedTestState SharedState { get; }
+
+        public RollForwardMultipleFrameworks(SharedTestState sharedState)
+        {
+            SharedState = sharedState;
+        }
+
+        // Soft roll forward from the inner framework reference [specified] to app's 5.1.1 (defaults)
+        [Theory]
+        [InlineData("5.0.0", Constants.RollForwardSetting.Disable, null)]
+        [InlineData("5.1.3", Constants.RollForwardSetting.Disable, "5.1.3")]
+        [InlineData("5.0.0", Constants.RollForwardSetting.LatestPatch, null)]
+        [InlineData("5.1.0", Constants.RollForwardSetting.LatestPatch, "5.1.3")]
+        [InlineData("5.0.0", null, "5.1.3")]
+        [InlineData("5.0.0", Constants.RollForwardSetting.Minor, "5.1.3")]
+        [InlineData("5.0.0", Constants.RollForwardSetting.LatestMinor, "5.1.3")] // The app reference which is Minor wins
+        [InlineData("1.0.0", Constants.RollForwardSetting.Minor, null)]
+        [InlineData("1.0.0", Constants.RollForwardSetting.Major, "5.1.3")]       // The app reference which is Minor wins
+        [InlineData("1.0.0", Constants.RollForwardSetting.LatestMajor, "5.1.3")] // The app reference which is Minor wins
```

Comment:
I would probably sort these in some logical way to make them easier to quickly grasp.  I find the current order erratic.

Might be nice to add 
+ `5.1.1` cases just to be complete.
+ `5.1.1-preview.1.0` cases
+ `ClosestPatch` cases if you took my earlier advice.

Reply:
This comment is probably somewhat incoherent, I looked at the test cases w/o fully looking at the test.